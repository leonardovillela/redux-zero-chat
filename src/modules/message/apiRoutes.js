export default {
  allMessagesFromDate(datetime) {
    return `/messages?createdAt_gte=${datetime}`
  },
  get createMessage() {
    return '/messages';
  },
}
