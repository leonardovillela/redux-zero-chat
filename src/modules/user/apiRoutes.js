export default {
  get index() { return '/users' },
  get lastUser() { return `${this.index}?_sort=id&_order=desc&_limit=1` },
};
