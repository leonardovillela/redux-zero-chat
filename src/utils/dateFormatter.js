export function format(date) {
  return new Date(parseInt(date)).toLocaleString();
}
