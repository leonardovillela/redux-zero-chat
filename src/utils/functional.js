export function compose(...args) {
  return args.reduce((accumulator, currentFunction) => (...params) => {
    const result = currentFunction(...params);
    return result.then ? result.then(accumulator) : accumulator(result);
  });
}
