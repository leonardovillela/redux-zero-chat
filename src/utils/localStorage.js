export function setJSONData(key, data) {
  localStorage.setItem(key, JSON.stringify(data));
  return data;
}
