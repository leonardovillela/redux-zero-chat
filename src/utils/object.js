import { mapValues, isFunction, isObject } from 'lodash';

export function generateNewObjectWithAllFunctionsInvoked(object) {
  return mapValues(object, value => isFunction(value)
      ? value()
      : isObject(value) ? this(object) : value
    );
}
