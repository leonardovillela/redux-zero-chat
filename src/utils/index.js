import * as dateFormatter from './dateFormatter';
import * as localStorage from './localStorage';
import * as object from './object';

export default {
  dateFormatter,
  localStorage,
  object,
};
