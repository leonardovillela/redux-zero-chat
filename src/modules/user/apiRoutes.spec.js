import apiRoutes from './apiRoutes';

import utils from '../../utils';

describe('Test API Routes unit', () => {
  it('Check if structure keeps equal', () => {
    expect(utils.object.generateNewObjectWithAllFunctionsInvoked(apiRoutes))
      .toMatchSnapshot();
  });
});
