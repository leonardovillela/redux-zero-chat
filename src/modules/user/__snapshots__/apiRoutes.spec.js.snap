// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test API Routes unit Check if structure keeps equal 1`] = `
Object {
  "index": "/users",
  "lastUser": "/users?_sort=id&_order=desc&_limit=1",
}
`;
